### PR TITLE
Alerting: Show legacy provisioned alert rules warning

### LINKED ISSUE #58169: Let users know and migrate their unused provisioned legacy alerts in frontend
Until recently, legacy alerting => Grafana Alerting was not allowing the migration of provisioned legacy alerts.

Even now, users on Grafana Alerting might have legacy alerts provisioned in dashboards that are useless. Those need to be able to be migrated.

This is not the backend issue https://github.com/grafana/grafana/issues/43546, only frontend.

More details needed.

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

This PR adds a warning when a user is browsing a dashboard that has legacy alert rules embedded in the panel configuration.

A helpful message is included and a link to the documentation (link to documentation TBD) on how to proceed to migrate.

image

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/58169

**Notes for reviewers**

Setting this to draft mode for now – ideally this would only show for users of Grafana 11+ and I have to figure out a good way to do that (or just wait to merge until after 10.4)

Additionally the copy needs to be reviewed too before this is good to go (cc @brendamuir)

### CODE REVIEW

**Comment by brendamuir** [comment]
Hey Gilles -

**Comment by konrad147** [comment]
I wonder if we should display a similar warning on the `JSON model` page or `Provisioned dashboard` summary.
The Alert rules button is a new thing. Users might overlook it, and errors could go unnoticed. 

Most users modifying provisioned dashboards use the `Provisioned dashboard` summary view to copy the JSON content

image

image

**Comment by gillesdemey** [comment]
Great idea! I'll see if I can move the fetching logic in to a wrapper component so we can have a portable component we can put anywhere we have a `dashboardUid`

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx [hunk 1]
```diff
@@ -179,7 +179,8 @@ export function InteractiveTable<TableData extends object>({
 
   const tableHooks: Array<PluginHook<TableData>> = [useSortBy, useExpanded];
 
-  const paginationEnabled = pageSize > 0;
+  const multiplePages = data.length > pageSize;
+  const paginationEnabled = pageSize > 0 && multiplePages;
```
Comment (gillesdemey) [inline packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx:183]: This changes seemed sensible for my use-case but maybe it was intended to also show the pagination controls even if the data set is smaller than the page size?

**Comment by ppcano** [comment]
This warning is shown whether the alert was previously migrated or not.

legacy_rules_in_dashboard

<br/><br/>

Clicking on the panel title will take the user directly to that panel, with the alert tab already open.


**1) Non-migrated alert**
  non_migrated_alert


**2) Migrated alert**
migrated_alert




The migrated alert case may be confusing:
- We inform the user that has legacy alerts but these were already migrated. 
- They may wonder, is the alert on the Alert tab the new or the old one?


I'm planning to document these various cases but wondered if there is anything else that could be done on the UI (cc @gillesdemey ).

File: public/app/features/alerting/unified/integration/AlertRulesDrawerContent.tsx [hunk 2]
```diff
@@ -27,12 +40,17 @@ export default function AlertRulesDrawerContent({ dashboardUid }: Props) {
     .flatMap((g) => g.rules)
     .filter((rule) => rule.annotations[Annotation.dashboardUID] === dashboardUid);
 
+  const loading = loadingRulesData;
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/integration/AlertRulesDrawerContent.tsx:43]: why do we need this new variable `loadingRulesData`, why don't we use directly loading from here?:
`const { loading } = useAsync(async () => {`

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! 🚀
Added a non blocking comment 👀

File: public/app/features/alerting/unified/utils/misc.ts [hunk 3]
```diff
@@ -153,8 +153,22 @@ export function makeDashboardLink(dashboardUID: string): string {
...
-export function makePanelLink(dashboardUID: string, panelId: string): string {
-  return createUrl(`/d/${encodeURIComponent(dashboardUID)}`, { viewPanel: panelId });
+type PanelLinkParams = {
+  viewPanel?: string;
+  editPanel?: string;
+  tab?: 'alert' | 'transform' | 'query';
+};
+export function makePanelLink(dashboardUID: string, panelId: string, queryParams: PanelLinkParams = {}): string {
+  const panelParams = new URLSearchParams({
+    viewPanel: panelId,
```
Comment (konrad147) [inline public/app/features/alerting/unified/utils/misc.ts:163]: nit: Doesn't `...queryParams` do the job?

File: public/app/features/alerting/unified/integration/AlertRulesDrawerContent.tsx (hunk 2 above)
Comment (konrad147) [reply public/app/features/alerting/unified/integration/AlertRulesDrawerContent.tsx:43]: Should we include loading state of `fetchDashboard` too?

File: public/app/features/alerting/unified/integration/LegacyAlertsDeprecationNotice.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  const panelsWithLegacyAlerts = getLegacyAlertPanelsFromDashboard(dashboard);
+
+  // don't show anything when the user has no legacy alerts defined
+  const hasLegacyAlerts = !isEmpty(panelsWithLegacyAlerts);
+  if (!hasLegacyAlerts) {
+    return null;
+  }
+
+  return dashboard.uid ? <LegacyAlertsWarning dashboardUid={dashboard.uid} panels={panelsWithLegacyAlerts} /> : null;
```
Comment (konrad147) [inline public/app/features/alerting/unified/integration/LegacyAlertsDeprecationNotice.tsx:31]: Can `dashboard.uid` be undefined?

File: public/app/features/alerting/unified/utils/misc.ts (hunk 3 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/utils/misc.ts:163]: Oh that's a good point – I think I refactored this and forgot to simplify this part :D

File: public/app/features/alerting/unified/integration/LegacyAlertsDeprecationNotice.tsx (hunk 4 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/integration/LegacyAlertsDeprecationNotice.tsx:31]: Yep, that's how it seems to be typed – I suspect the same model is used for existing dashboard as for dashboards that are being created and don't have a UID from the database yet.

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**